pkg/kubelet/config/config.go

```diff
@@ -94,6 +94,8 @@ func (c *PodConfig) SeenAllSources(seenSources sets.String) bool {
 	if c.pods == nil {
 		return false
 	}
+	c.sourcesLock.Lock()
```

Comment:
Is there a test case or test failure demonstrating this race condition?  Would be nice to have one showing it that this change causes to pass (especially with `-race`)

Reply:
Note just by cursory examination the lock guarding mutation also has to guard read and so this appears to be correct, I just want to be certain we squash it AND we don't miss something else here.

Reply:
The [unit test](https://github.com/kubernetes/kubernetes/pull/106551#issuecomment-974236664) may be can demonstrate this race condition. There are "WARNING: DATA RACE" information when I run the test with `-race`, and the test is passed when the `c.sourcesLock.Lock()` added.